### PR TITLE
Removing unused packaging code and update README

### LINKED ISSUE #155: Circular reference in Windows 'make.ps1 New-MSI' command.
When executing the [Windows makefile](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/blob/9bfb379327ffc7770ee8d44eeff4c871f4a98d97/.build/msi/make.ps1#L60), an error is created indicating the .msi creation process needs access to the exe.

```C:\otel\.build\msi\google-cloud-metrics-agent.wxs(28) : error LGHT0103 : The system cannot find the file './bin/google-cloud-metrics-agent_windows_amd64.exe'.```

[Looks like part of the service definition is what is causing the problem.](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/blob/9bfb379327ffc7770ee8d44eeff4c871f4a98d97/.build/msi/google-cloud-metrics-agent.wxs#L28)

Unfortunately the reason I am executing the makefile is as a means to create the most up-to-date .exe, creating somewhat of a circular dependency.

### PR DESCRIPTION
**Pull request by braydonk**
This repo was originally designed for something completely different than it's used for today. It would have been incredibly misleading for anyone discovering the repo without any context. This PR is to completely gut all of the unused packaging code and make targets, as well as revamp the README entirely to reflect what this collector is actually used for today.

Closes #155

### CODE REVIEW

File: Makefile
Comment (igorpeshansky) [inline Makefile:41]: Should this not become `presubmit` again?

File: Makefile
Comment (igorpeshansky) [inline Makefile:52]: I'm a bit confused. This is already installed as part of `install-tools` — why install it twice?

File: Makefile
Comment (igorpeshansky) [inline Makefile:62]: The original code ran each of these as a separate command in the `Makefile`, which would stop the build if any of them failed. The current code runs them unconditionally, so only the status of the last command would determine the result of the entire group.

On the flip side, doesn't `go install` allow multiple modules? Why not make it a single `go install` command instead?

File: Makefile
Comment (igorpeshansky) [inline Makefile:65]: Are we upgrading from `mdatagen@0.67.0` to, what, `0.74.0`? Is that significant?

File: Makefile
Comment (igorpeshansky) [inline Makefile:99]: Trailing whitespace?

File: Makefile
Comment (igorpeshansky) [inline Makefile:44]: Trailing whitespace?

File: Makefile
Comment (igorpeshansky) [inline Makefile:53]: If this command fails, it would be hard to figure out whether `go get` or `go mod tidy` failed. Keeping them as two separate commands would allow `make` to report the failure appropriately…

File: Makefile
Comment (igorpeshansky) [inline Makefile:118]: This is essentially the same command as in the `test` target, but without the `-v`. I wonder if it makes sense to define one in terms of another, e.g.:
```
.PHONY: test
test:
	go test -tags=$(GO_BUILD_TAGS) $(VERBOSE_TESTS) -race ./...

.PHONY: test_verbose
test_verbose:
	$(MAKE) VERBOSE_TESTS=-v test
```

File: Makefile
Comment (igorpeshansky) [inline Makefile:110]: This is essentially the same command as in the `build` target, but with a different binary name. I wonder if it makes sense to factor out the common command, e.g.:
```
.PHONY: build-impl
build-impl:
ifndef BINARY_OUT
	$(error "BINARY_OUT is undefined")
endif
	go build -tags=$(GO_BUILD_TAGS) -o $(BINARY_OUT) $(LD_FLAGS) ./cmd/otelopscol

GO_BUILD_OUT ?= ./bin/otelopscol
.PHONY: build
build:
	$(MAKE) BINARY_OUT=$(GO_BUILD_OUT) build-impl

OTELCOL_BINARY = google-cloud-metrics-agent_$(GOOS)_$(GOARCH)$(EXTENSION)
.PHONY: build_full_name
build_full_name:
	$(MAKE) BINARY_OUT=./bin/$(OTELCOL_BINARY) build-impl
```

**Review by igorpeshansky — COMMENTED** [review]
One actual bug (`s/ci/presubmit/`)…

File: .build/Dockerfile
Comment (braydonk) [reply .build/Dockerfile:3]: It was here when I got here, I thought it seemed unnecessary but I didn't want to touch it since I wasn't sure. I can get rid of it

File: Makefile
Comment (braydonk) [reply Makefile:52]: This is specifically to update the dependency in the `go.mod`. This needs to update in sync with the rest of the opentelemetry dependencies, which has been missed in the last couple upgrades. This will add it to the automation that updates all the other otel deps, and I'm adding a step in the guide to regenerate all the metadata with the new version of the library too.

File: Makefile
Comment (braydonk) [reply Makefile:62]: I didn't know `go install` supported multiple at once. I'll switch it to that.

File: Makefile
Comment (braydonk) [reply Makefile:65]: It's been added in `internal/tools/go.mod`, it will be updated to 0.71.0 in this PR, but I will probably rebase this PR on the 0.74.0 upgrade PR once that's merged and once that's done it will be up to 0.74.0.

File: Makefile
Comment (braydonk) [reply Makefile:53]: Doing a `cd` requires that they all be one command right? I thought running a `cd` as a separate command in a make target won't actually change the whole directory context.

File: Makefile
Comment (braydonk) [reply Makefile]: Did a version of this in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: Makefile
Comment (braydonk) [reply Makefile]: Done in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: Makefile
Comment (braydonk) [reply Makefile]: Fixed in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: Makefile
Comment (braydonk) [reply Makefile]: Fixed in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: Makefile
Comment (braydonk) [reply Makefile]: Done in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: Makefile
Comment (braydonk) [reply Makefile]: Fixed in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

File: .build/Dockerfile
Comment (braydonk) [reply .build/Dockerfile]: Removed in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

Approving with a couple of optional suggestions.

File: Makefile
Comment (braydonk) [reply Makefile]: I separated the update and tidy into two commands as suggested. For collapsing the tidy into the update loop, after looking at it I realize that the old `update-components` make target needs to be re-done anyway. It doesn't catch all opentelemetry dependencies right now. So I will probably be rewriting a chunk of that in another PR anyway.

File: docs/dev/upgrade-opentelemetry.md
Comment (braydonk) [reply docs/dev/upgrade-opentelemetry.md:5]: I collapsed the update, install, and generate into a single make target. Since on most OpenTelemetry upgrades build and test fail by default anyway, I decided to leave those targets separate since they will likely need lots of manual work anyway to get each one to pass.

File: docs/dev/upgrade-opentelemetry.md
Comment (igorpeshansky) [inline docs/dev/upgrade-opentelemetry.md:12]: "details of the actual details"? I think you meant `s/for details of/for/`…
```suggestion
Read on for the actual details of the target, or skip ahead to [Test](#test) to continue the process.
```

File: docs/dev/upgrade-opentelemetry.md
Comment (igorpeshansky) [inline docs/dev/upgrade-opentelemetry.md:18]: "dependencies include".
```suggestion
Those dependencies include the `mdatagen` tool, which is in a separate place from libraries (read more in [tools.md](./tools.md)). Since the `mdatagen` version has been updated in the tools `go.mod`, re-install tools to actually install the new version:
```

File: docs/dev/upgrade-opentelemetry.md
Comment (igorpeshansky) [inline docs/dev/upgrade-opentelemetry.md:12]: [Optional] Want to make it a collapsible zippy section (see, e.g., https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab)?

File: Makefile
Comment (igorpeshansky) [inline Makefile:58]: I don't know that setting a variable in a separate command actually affects the subsequent commands — did you want to set this for one of the sub-commands instead?
```suggestion
	$(MAKE) update-components 
	$(MAKE) install-tools 
	$(MAKE) GO_BUILD_TAGS=gpu generate
```

File: docs/dev/upgrade-opentelemetry.md
Comment (braydonk) [reply docs/dev/upgrade-opentelemetry.md]: Deleted this line when I changed it to a zip section

File: docs/dev/upgrade-opentelemetry.md
Comment (braydonk) [reply docs/dev/upgrade-opentelemetry.md]: Fixed

File: Makefile
Comment (braydonk) [reply Makefile]: You're right, fixed it

File: docs/dev/upgrade-opentelemetry.md
Comment (braydonk) [reply docs/dev/upgrade-opentelemetry.md]: Great idea, this is done

File: docs/dev/upgrade-opentelemetry.md
Comment (igorpeshansky) [inline docs/dev/upgrade-opentelemetry.md:13]:
```suggestion
    <summary>`update-opentelemetry` target details</summary>
```

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

One minor formatting suggestion. Please feel free to submit after that.

File: docs/dev/upgrade-opentelemetry.md
Comment (igorpeshansky) [reply docs/dev/upgrade-opentelemetry.md]: Oops, should have tested this suggestion, sorry. Turns out the `<summary>` tag does not support inner markdown. Sending #161 to fix.

**Pull request merged**
